usaspending_api/wsgi.py

```diff
@@ -7,10 +7,82 @@
 https://docs.djangoproject.com/en/2.2/howto/deployment/wsgi/
 """
 
+# Standard library imports
 import os
 
+# Django imports
 from django.core.wsgi import get_wsgi_application
 
+# OpenTelemetry imports
+from opentelemetry import trace
+from opentelemetry.instrumentation.django import DjangoInstrumentor
+from opentelemetry.instrumentation.wsgi import OpenTelemetryMiddleware
+
+# Local imports
+from usaspending_api.common.logging import configure_logging
+
+# Constants
+HEADERS_TO_CAPTURE = [
+    "content-length",
+    "content-type",
+    "host",
+    "origin",
+    "referer",
+    "ua-is-bot",
+    "user-agent",
+    "x-forwarded-for",
+    "x-requested-with",
+    "allow",
+    "cache-trace",
+    "is-dynamically-rendered",
+    "key",
+    "strict-transport-security",
+]
+
+
+def request_hook(span, environ):
+
+    if span and span.is_recording():
+        for header in HEADERS_TO_CAPTURE:
+            header_env_name = f"HTTP_{header.replace('-', '_').upper()}"
+            header_value = environ.get(header_env_name)
+            if header_value:
+                span.set_attribute(f"http.request.header.{header}", header_value)
+
+    if os.getenv("USASPENDING_DB_HOST") == "127.0.0.1" and os.getenv("TOGGLE_OTEL_CONSOLE_LOGGING") == "True":
```

Comment:
Could be a good spot to use `settings.IS_LOCAL` if Django is started by this point. If it is not, then we need to find a way to better capture that this is running on a local build.